### PR TITLE
Separate turf flags from flags_1

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

Ports:
- https://github.com/tgstation/tgstation/pull/57263

Separates turf_flags bitfield from flags_1

## Why It's Good For The Game

Cleaner code, less bloated bitfields

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Screenshot 2024-04-04 112024

</details>

## Changelog
:cl: rkz, Fikou
code: separates turf_flags into its own bitfield
/:cl:

### CODE REVIEW

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -657,7 +657,7 @@
 				if(phase_state)
 					flick(phase_state, src)
 				var/turf/target = get_step(src, dir)
-				if(target.flags_1 & NOJAUNT_1)
+				if(target.flags_1 & NOJAUNT)
```
Comment (EvilDragonfiend) [inline code/game/mecha/mecha.dm]: It still uses flags_1

File: code/game/turfs/open/reebe_void.dm [hunk 2]
```diff
@@ -6,7 +6,7 @@
 	planetary_atmos = TRUE
 	bullet_bounce_sound = null //forever falling
 	tiled_dirt = FALSE
-	flags_1 = NOJAUNT_1
+	flags_1 = NOJAUNT
```
Comment (EvilDragonfiend) [inline code/game/turfs/open/reebe_void.dm:9]: *shrug

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
I guess you need to recheck some

File: code/game/turfs/open/reebe_void.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/game/turfs/open/reebe_void.dm:9]: intended

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/game/mecha/mecha.dm]: Good catch. Fixed in fe35981

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
Would you please take a look into your PR again? You are not separating flags correctly. The ones I told wasn't what really all.
You made 4 turf flags
```DM
#define NOJAUNT (1<<0)
#define UNUSED_RESERVATION_TURF (1<<1)
#define NO_LAVA_GEN (1<<3) //! Blocks lava rivers being generated on the turf
#define NO_RUINS (1<<4) //! Blocks ruins spawning on the turf
```
and these are supposed to go in `turf_flags` but still you're putting it into `flags_1`

**Pull request closed without merging**